### PR TITLE
Data structures via the cli supporting docs

### PR DESCRIPTION
**Pull request by jamessnowplow**
work in progress

### CODE REVIEW

File: docs/understanding-tracking-design/managing-your-data-structures/cli/index.md
Comment (gleb-lobov) [inline docs/understanding-tracking-design/managing-your-data-structures/cli/index.md:8]: Should cloud be included as well?

File: docs/understanding-tracking-design/managing-your-data-structures/cli/index.md
Comment (gleb-lobov) [inline docs/understanding-tracking-design/managing-your-data-structures/cli/index.md:108]: Probably link needs host update

**Review by gleb-lobov — APPROVED** [review]
Looks good overall, but I think we don't communicate anywhere that you need to edit the fiels in place, instead of creating new files for each version. It's kinda implied by names not having the versions, but I think we still want to mention it somewhere. But maybe the recipe will have it

File: docs/understanding-tracking-design/managing-your-data-structures/cli/index.md
Comment (jamessnowplow) [reply docs/understanding-tracking-design/managing-your-data-structures/cli/index.md:8]: as far as i can tell this is just how the docs differentiate between BDP and Community. There are no other options.

Comment (gleb-lobov) [reply docs/understanding-tracking-design/managing-your-data-structures/cli/index.md:8]: Maybe I'm mixing something up, or the docs were changed since the last time I've done anything there

File: docs/recipes/recipe-data-structures-in-git/index.md
Comment (gleb-lobov) [inline docs/recipes/recipe-data-structures-in-git/index.md:25]: I think the product is called `github actions`, but Im not sure what's the difference

Comment (jamessnowplow) [reply docs/recipes/recipe-data-structures-in-git/index.md:25]: I thought that too but their docs are a bit confusing. They refer to workflows in a lot of places then actions in their more detailed stuff.

File: docs/recipes/recipe-data-structures-in-git/index.md
Comment (gleb-lobov) [inline docs/recipes/recipe-data-structures-in-git/index.md:229]: Should we move the `--managed-from` to be an env variable as well?

File: docs/recipes/recipe-data-structures-in-git/index.md
Comment (gleb-lobov) [reply docs/recipes/recipe-data-structures-in-git/index.md:25]: Don't know, need some marketing people. I would use actions since it's a more recognisable name

File: docs/recipes/recipe-data-structures-in-git/index.md
Comment (jamessnowplow) [reply docs/recipes/recipe-data-structures-in-git/index.md:229]: maybe? I don't think I feel strongly. It kind of does fit with things like org id, yeah, id be ok with it either way

File: docs/recipes/recipe-data-structures-in-git/index.md
Comment (jamessnowplow) [reply docs/recipes/recipe-data-structures-in-git/index.md:25]: As far as I can tell, actions is the platform and workflows are the things you create. Yeah, I can change it to actions to avoid anyone who hasn't read the definitions getting confused.

File: docs/recipes/recipe-data-structures-in-git/index.md
Comment (gleb-lobov) [reply docs/recipes/recipe-data-structures-in-git/index.md:229]: I think it makes sense, it makes the command a bit cleaner

File: docs/recipes/recipe-data-structures-in-git/index.md
Comment (cksnp) [inline docs/recipes/recipe-data-structures-in-git/index.md:60]: This paragraph is unclear to me.

File: docs/recipes/recipe-data-structures-in-git/index.md
Comment (cksnp) [inline docs/recipes/recipe-data-structures-in-git/index.md:137]: I found that TM superscript funny but can we please remove it from the docs :)

File: docs/recipes/recipe-data-structures-in-git/index.md
Comment (cksnp) [inline docs/recipes/recipe-data-structures-in-git/index.md:192]:
```suggestion
We'll not go into the details of creating github repositories and initial commits here, the [github docs](https://docs.github.com/) provide full guidance how to do those. The next few steps will assume a working github repository containing the directory and data structure we created in the previous section. It will have two branches named `main` and `develop` which should be in sync.
```

File: docs/recipes/recipe-data-structures-in-git/index.md
Comment (cksnp) [inline docs/recipes/recipe-data-structures-in-git/index.md:259]: `pub`? Not `published`? I assume the arg parse tool may allow that, but let's settle on `publish` as the clearer option.

File: docs/recipes/recipe-data-structures-in-git/index.md
Comment (cksnp) [inline docs/recipes/recipe-data-structures-in-git/index.md:273]: Probably `validate` instead of `publish`?

File: docs/recipes/recipe-data-structures-in-git/index.md
Comment (cksnp) [inline docs/recipes/recipe-data-structures-in-git/index.md:320]:
```suggestion
Validation has failed. To identify the problem we open the 'file' tab on the pull request and see..
```

File: docs/recipes/recipe-data-structures-in-git/index.md
Comment (cksnp) [inline docs/recipes/recipe-data-structures-in-git/index.md:353]:
```suggestion
Now that validation has passed, our colleagues can feedback on our changes and if everyone is happy we can merge to `develop` which will trigger our `publish-develop.yml` workflow.
```

File: docs/recipes/recipe-data-structures-in-git/index.md
Comment (cksnp) [inline docs/recipes/recipe-data-structures-in-git/index.md:364]: I would completely remove this section.

File: docs/understanding-tracking-design/managing-your-data-structures/cli/index.md
Comment (cksnp) [inline docs/understanding-tracking-design/managing-your-data-structures/cli/index.md:22]: h4s render as simple text, so perhaps make them bold too? (not sure if that works)

File: docs/understanding-tracking-design/managing-your-data-structures/cli/index.md
Comment (cksnp) [inline docs/understanding-tracking-design/managing-your-data-structures/cli/index.md:33]: As above

**Review by cksnp — CHANGES_REQUESTED** [review]
I have a few small suggestions (and I think a typo fix or two).
Before we merge to prod I need to coordinate with marketing, support, etc. So let's have it ready without merging.

File: docs/recipes/recipe-data-structures-in-git/index.md
Comment (jamessnowplow) [reply docs/recipes/recipe-data-structures-in-git/index.md:364]: Both or just the 'do next'? I was trying to stay consistent with the other recipes but it is a bit redundant.

File: docs/understanding-tracking-design/managing-your-data-structures/cli/index.md
Comment (jamessnowplow) [reply docs/understanding-tracking-design/managing-your-data-structures/cli/index.md:22]: The styles changed under me :) (for the better) I'll try and make it look better.

File: docs/recipes/recipe-data-structures-in-git/index.md
Comment (jamessnowplow) [reply docs/recipes/recipe-data-structures-in-git/index.md:137]: wrong tone, understood.

File: docs/recipes/recipe-data-structures-in-git/index.md
Comment (jamessnowplow) [reply docs/recipes/recipe-data-structures-in-git/index.md:60]: I was trying to get across the pattern but also that you don't have to follow it. Maybe that isn't important to note. I'll reword

File: docs/recipes/recipe-data-structures-in-git/index.md
Comment (jamessnowplow) [reply docs/recipes/recipe-data-structures-in-git/index.md:192]: At primary school I got roasted for writing extremely long sentences and using commas when I should have used full stops. That roasting has lived with me ever since and now every time I feel like I want to comma I find a way to full stop.

**Review by cksnp — APPROVED** [review]
LG! As said, let's wait for 👍 by marketing etc before merging

**Comment by johnmicahreid** [comment]
@jamessnowplow can we add an explicit example where you request someone else's approval on a new data structure e.g. a data team member? I know it's implied by managing stuff via Github, but it would be really good to call out an approval workflow specifically (e.g. where Contributor A does not have merge approval and Reviewer B does)

**Comment by jamessnowplow** [comment]
@johnmicahreid I'm a bit worried about accidentally writing more about github setup and workflows than snowplow-cli.

What level of knowledge are we assuming here? Would the example include how to setup approval workflows and branch rules and prs in github? Or try to capture the essence in a series of screenshots or something?

**Pull request merged**
